### ISSUE TITLE
How to access the response headers?

### ISSUE THREAD

**Issue by luzfcb** [vintasoftware/tapioca-wrapper#171]
Hi, I'm maintaining an existing application from a customer and there is an API client implementation of an external service that is implemented using tapioca-wrapper (v1.4.2). The external service now includes some headers in the response of each request and I need to perform different actions depending on the value that specific headers have.

Is there an official way to access the raw response object (not only the response body) from the `TapiocaClient` or `TapiocaClientExecutor`?

From the example below, how to access the response HEADERS from the `tapioca_client` instance, `tapioca_client_executor` instance, or from the result object of the `get()` method call?

```python
from tapioca import TapiocaAdapter, generate_wrapper_from_adapter, JSONAdapterMixin
from requests.auth import HTTPBasicAuth

from .mapping import RESOURCE_MAPPING


class FooBarClientAdapter(JSONAdapterMixin, TapiocaAdapter):
    api_root = 'https://api.foobar.com/'
    resource_mapping = RESOURCE_MAPPING


FooBarClientClass = generate_wrapper_from_adapter(FooBarClientAdapter)

tapioca_client = FooBarClientClass()

url = "https://api.foobar.com/resource/123/?x=1&y=2"

tapioca_client_executor = tapioca_client._wrap_in_tapioca_executor(url)

page = 1

data = tapioca_client_executor.get(params={'page': page})


```

**Comment by fjsj** [comment]
Have you tried accessing the response from the executor? We have a property there:
https://github.com/vintasoftware/tapioca-wrapper/blob/edcbe115df61e09350a8e408653fa9252aade9cb/tapioca/tapioca.py#L209-L212

So `data().response` is probably the right answer here.

To get the headers: `data().response.headers`.
Just tested with `tapioca-github` on gists endpoint and it worked fine.

**Comment by luzfcb** [comment]
@fjsj Thanks a lot. This is a good candidate for a new Cookbook section in the documentation. The current documentation is neither nearly clear nor straightforward on that particular subject.

**Comment by luzfcb** [comment]
Anyone who has the same doubts, this is a 100% working example.
Note: I used https://designer.mocky.io/design to create a mock response that includes a header.

```python
from tapioca import TapiocaAdapter, generate_wrapper_from_adapter, JSONAdapterMixin


RESOURCE_MAPPING = {}

class FooBarClientAdapter(JSONAdapterMixin, TapiocaAdapter):
    api_root = 'https://run.mocky.io'
    resource_mapping = RESOURCE_MAPPING


FooBarClientClass = generate_wrapper_from_adapter(FooBarClientAdapter)

tapioca_client = FooBarClientClass()

url = "https://run.mocky.io/v3/927b0e28-7ec2-44fc-ab3a-d4bd386ad130"

tapioca_client_executor = tapioca_client._wrap_in_tapioca_executor(url)

new_tapioca_client_class = tapioca_client_executor.get(params={'page': 1})
new_tapioca_client_executor = new_tapioca_client_class()
print(new_tapioca_client_executor.response.headers)

```

**Comment by fjsj** [comment]
> This is a good candidate for a new Cookbook section in the documentation. The current documentation is neither nearly clear nor straightforward on that particular subject.

Agreed @luzfcb , I'll leave the issue open due to that. Please feel free to open a PR for this additional documentation.
